src/repositories/urlRepository.js

```diff
@@ -0,0 +1,58 @@
+import { ShortUrl } from '../models/ShortUrl.mjs';
+import { Clicks } from '../models/Clicks.mjs';
+import { User } from '../models/User.mjs';
+
+export class UrlRepository {
+    async createUrl(urlData) {
+        const { user_id, short_url, original_url } = urlData;
+
+        return await ShortUrl.transaction(async (trx) => {
+            const user = await User.query().findById(user_id);
+
+            if (user == undefined) {
+                return false;
+            }
+
+            return await ShortUrl.query().insert({
+                original_url,
+                short_url,
+                user_id
+            });
+        });
+    }
+
+    async findUrlByShortCode(urlData) {
+        const { ip_address, shortCode } = urlData;
+        const short_url = 'http://localhost:3000/' + shortCode;
+
+        return await ShortUrl.transaction(async (trx) => {
+            const url = await ShortUrl.query().findOne({ short_url: short_url });
+
+            if (url == undefined) {

```

Comment:
é sempre bons utilizarmos o strict equality, isso evita alguns bugs chatos de javascript

```suggestion
            if (url === undefined) {

```

além disso, acho que não há problema em tratar como `if(!url)`, mas fica a seu critério sobre qual cê acha mais claro